src/components/Marker.tsx

```diff
@@ -91,26 +90,28 @@ export default function Marker({
     { name: 'select', handler: onSelect },
     { name: 'deselect', handler: onDeselect },
     { name: 'drag-start', handler: onDragStart },
```

Comment:
Don't you think it could also be helpful to be able to access the position onDragStart? Maybe even on select/deselect, although I could currently not think of any implementation where this would be needed. 

Of course all this would be able to be worked around, though in my opinion I would provide as much information with the events as we have available, as for some implementations this might make things a lot easier. I always hate it if the library I am using not provides the data with the events which I do need for some implementation.

I would even consider handing over the full marker object instead of only the coordinates as I feel the library should not limits the possible implementations by the user.

Reply:
This would feel less idiomatic in React, as it is information that the user already has access to (since they’re already providing it directly if they’re using the API correctly).

If there’s a need to get the full marker object, we could support this through refs (in the same way as `Map` exposes its ref). I would prefer to avoid exposing the MapKit JS objects in other places as it makes the API too specific to MapKit JS, which I would like to avoid.

Reply:
Hi @Nicolapps, 

I was thinking a lot about this and the more I am thinking about it the less sense it makes for me to handle the `drag-end` different than the other events. Sorry to say that.

1. Why would we want to force the user to add helper states (the user will always be required to listen the `drag-end`, although the user might not even be interested in the `drag-end` event, if for example the user would like to access the coordinates on `select` or even `drag-start`). Helper states add complexity in the code and reduce maintainability.
2.  Why would we add additional complexity to the library when it is not needed. Handling the events different, makes it more challenging to maintain, while we would have access to the coordinates from all the events without any additional effort.

Generally speaking, whenever there's some information available to use directly from the event (currently talking about the coordinates rather than the full JS object) I would always prefer using them over those from a helper state or ref.

Talking about exposing the MapKit JS objects, this is something I can follow on, though still would consider adding the data object (and maybe even some other raw data (not the full JS context) from the event) as a additional param in a later PR.

Reply:
Thank you for your replies, @nikischin! I’m not sure of when it would be helpful for the user to receive the current position in the other event handlers. I would expect the user to already have access to it (since the event handler is written at the same place where the marker position is set).

```
function MyMarker() {
  const [position, setPosition] = useState({ ... });
  return (
    <Marker
      latitude={position.latitude}
      longitude={position.longitude}
      onDragStart={() => {
        // use position
      }}
    />
  );
}
```

Do you have an example in your project where doing it as in this example isn’t feasible/practical?

Including the marker position where it isn’t needed would be confusing in my opinion (especially because the event handler for dragEnd would look similar to all the other handlers, while it is intrinsically different).

Reply:
@Nicolapps honestly, when providing a React library for a JavaScript library I would expect to have the same functionality as the JavaScript functionality and not gonna discuss wether a function in the JavaScript library is beneficial or not if it is easy to implement and maintain. I personally would expect the React library to be as close to the JavaScript library as it is possible and does make sense.

Also honestly speaking, I would never implement a library which doesn't include all the features I need and introduce too many workarounds for things which could be solved by the library super easily. In this case I would rather do my own custom implementation without a library where not I have the risk at some point it will not be maintained anymore and might add even more workarounds than I already know about.

So yes I could definitely make up a example where we wouldn't even have the coordinates in state as we might get them via props and pass them in a map function. Definitely I could introduce helper states to manage the marker positions and update them, though I really don't get why the library should act as guardian for the user not to use the position the other events would already provide. 

Also we implement the coordinates as first param and not as object, which definitely I can follow your arguments on. However if at some point we would wanna add the data attribute the library provides for easier context handling (yes this can be also done via state, though again adds additional complexity for the user) we would have the data attribute as the first parameter at all the functions except the drag-end where it would be second? This really would confuse me a lot.

And talking about the drag-end event I don't feel this is any different than the other events. It just provides the current position from the marker ref at the event time which in this case as of the nature of the event is updated already. Not sure if you are mixing this up with the dragging event which does have another implementation by the mapkit JS library already and does as the only event always provide the current in time updated coordinates as separate param which does differ from the current marker ref position. For all the other events you could also save the marker in a ref and access its coordinates (which in my opinion would end in a super ugly implementation for a user of this library).

And to sum up all this you have to think about this projects vision and decide whether you want it to be a shared codebase of different projects (which justifies asking me how I would need to implement features in my project) or a library which provides a set of features but doesn't predefine what functionality the user will make of it. Not saying though that for some heavy functionality you first might wanna have some need for it, though the functionality added should always provide a general solution and not just one to those who requested it.

Reply:
And speaking about event parameters, I would also consider adding the (React, not plain JS) ref of the current marker as a event parameter. It can make implementations so much easier for the user being able to directly access all the refs parameters within the events. Helper states again would solve this thing though can add quite a bit of complexity when combined with useMemos and other stuff and can at some point force re-renders at points where they would not be needed and therefore result in worse performance.